### PR TITLE
Refactor water window into Window class #13815

### PR DESCRIPTION
**Pull request by Gecko05**
This PR addresses #13815 

Refactored the code from water window into a new class called WaterWindow which inherits from Window class and overrides the On methods.
Added a new generic method for WindowFocusOrCreate that additionally takes position as an argument to create a window.

### CODE REVIEW

**Review by Gymnasiast — COMMENTED** [review]
Tested it. Seems to work correctly. I'll leave checking the code to @IntelOrca .

File: src/openrct2-ui/windows/Water.cpp [hunk 1]
```diff
@@ -37,182 +37,148 @@ static rct_widget window_water_widgets[] = {
...
-    // Draw number for tool sizes bigger than 7
-    if (gLandToolSize > MAX_TOOL_SIZE_WITH_SPRITE)
+    void OnDraw(rct_drawpixelinfo& dpi) override
     {
-        gfx_draw_string_centred(
-            dpi, STR_LAND_TOOL_SIZE_VALUE, screenCoords - ScreenCoordsXY{ 0, 2 }, COLOUR_BLACK, &gLandToolSize);
+        auto screenCoords = ScreenCoordsXY{ windowPos.x + window_water_widgets[WIDX_PREVIEW].midX(),
+                                            windowPos.y + window_water_widgets[WIDX_PREVIEW].midY() };
+
+        DrawWidgets(dpi);
```
Comment (JoelLarson) [inline src/openrct2-ui/windows/Water.cpp:145]: I believe this should be `DrawWidgets(&dpi);` with the change from *dpi to &dpi

File: src/openrct2-ui/windows/Water.cpp [hunk 2]
```diff
@@ -37,182 +37,148 @@ static rct_widget window_water_widgets[] = {
...
+        {
+            Close();
+        }
+    }
 
-/**
- *
- *  rct2: 0x006E6AB8
- */
-static void window_water_invalidate(rct_window* w)
```
Comment (JoelLarson) [inline src/openrct2-ui/windows/Water.cpp:179]: Is `window_water_invalidate()` mapped to `OnPrepareDraw()` ? I wasn't sure what to do with this in my scenery window PR.

File: src/openrct2-ui/windows/Water.cpp (hunk 1 above)
Comment (JoelLarson) [reply src/openrct2-ui/windows/Water.cpp:145]: I just noticed this was changed from `WindowDrawWidgets()`. My comment is invalid.

File: src/openrct2-ui/windows/Water.cpp (hunk 2 above)
Comment (duncanspumpkin) [reply src/openrct2-ui/windows/Water.cpp:179]: Yes we renamed invalidate to prepare draw to better reflect what it is being used for

**Review by IntelOrca — APPROVED** [review]

**Pull request merged**